plugins/module_utils/network/nxos/config/ospfv2/ospfv2.py

```diff
@@ -0,0 +1,232 @@
+    dict_merge,
+)
+from ansible_collections.cisco.nxos.plugins.module_utils.network.nxos.facts.facts import (
+    Facts,
+)
+from ansible_collections.cisco.nxos.plugins.module_utils.network.common.resource_module import (
+    ResourceModule,
+)
+from ansible_collections.cisco.nxos.plugins.module_utils.network.common.rm_utils import (
+    get_from_dict,
+)
+
+
+class Ospfv2(ResourceModule):
+    """
+    The nxos_ospfv2 class
+    """
+
+    def __init__(self, module):
+        super(Ospfv2, self).__init__(
+            empty_fact_val={},
+            facts_module=Facts(module),
+            module=module,
+            resource="ospfv2",
+            tmplt=Ospfv2Template(),
+        )
+        self.parsers = [
+            "router_id",
+            "auto_cost",
+            "graceful_restart.set",
+            "graceful_restart.helper_disable",
+            "graceful_restart.planned_only",
+            "isolate",
+            "log_adjacency_changes",
+            "max_lsa",
+            "mpls.traffic_eng.router_id",
+            "mpls.traffic_eng.multicast_intact",
+            "name_lookup",
+            "passive_interface.default",
+            "rfc1583compatibility",
+            "shutdown",
+            "default_information.originate",
+            "default_metric",
+            "distance",
+            "table_map",
+            "timers.lsa_arrival",
+            "timers.lsa_group_pacing",
+            "timers.throttle.lsa",
+            "timers.throttle.spf",
+            "maximum_paths",
+            "max_metric",
+            "down_bit_ignore",
+            "capability.vrf_lite",
+            "bfd",
+        ]
+
+    def execute_module(self):
+        """ Execute the module
+
+        :rtype: A dictionary
+        :returns: The result from module execution
+        """
+        if self.state not in ["parsed", "gathered"]:
+            self.gen_config()
+            self.run_commands()
+        return self.result
+
+    def gen_config(self):
+        """ Select the appropriate function based on the state provided
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        wantd = {
+            (entry["process_id"]): entry
+            for entry in self.want.get("processes", [])
+        }
+        haved = {
+            (entry["process_id"]): entry
+            for entry in self.have.get("processes", [])
+        }
+
+        # turn all lists of dicts into dicts prior to merge
+        for entry in wantd, haved:
+            self._ospf_list_to_dict(entry)
+
+        # if state is merged, merge want onto have and then compare
+        if self.state == "merged":
+            wantd = dict_merge(haved, wantd)
+
+        # if state is deleted, empty out wantd and set haved to wantd
+        if self.state == "deleted":
+            haved = {k: v for k, v in haved.items() if k in wantd or not wantd}
```

Comment:
also, we need to change all of the `.items()` to ansible `iteritems()` instead
```suggestion
            haved = {k: v for k, v in iteritems(haved) if k in wantd or not wantd}
```

Reply:
Ah, right.